src/accumulator.rs

```diff
@@ -0,0 +1,125 @@
+//! An accumulator used to collect chunked COBS data and deserialize it.
+
+use core::mem::MaybeUninit;
+use serde::Deserialize;
+
+/// An accumulator used to collect chunked COBS data and deserialize it.
+pub struct CobsAccumulator<const N: usize> {
+    buf: [u8; N],
+    idx: usize,
+}
+
+/// The result of feeding the accumulator.
+pub enum FeedResult<'a, T> {
+    /// Consumed all data, still pending.
+    Consumed,
+
+    /// Buffer was filled. Contains remaining section of input, if any.
+    OverFull(&'a [u8]),
+
+    /// Reached end of chunk, but deserialization failed. Contains remaining section of input, if.
+    /// any
+    DeserError(&'a [u8]),
+
+    /// Deserialization complete. Contains deserialized data and remaining section of input, if any.
+    Success {
+        /// Deserialize data.
+        data: T,
+
+        /// Remaining data left in the buffer after deserializing.
+        remaining: &'a [u8],
+    },
+}
+
+impl<const N: usize> CobsAccumulator<N> {
+    /// Create a new accumulator.
+    pub fn new() -> Self {
+        CobsAccumulator {
+            buf: unsafe { MaybeUninit::zeroed().assume_init() },
```

Comment:
```suggestion
            buf: [0; N],
```

Reply:
Ooo shiny. I just pushed a `fixup!` commit. Should I keep doing that so you can rebase, or are you going to just squash and merge?

Reply:
You can just discard this comment, you already did it (don't forget to remove the `use` too!